### PR TITLE
Add climate preset tile feature

### PR DESCRIPTION
**Pull request by Weissnix4711**
## Proposed change

Adds climate preset mode feature to tile card.

Preview 1

Preview 2

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by piitaya** [comment]
Hello 👋 Thank you for the contribution.
Preset can be custom so we will not have icon for every preset so I don't think we can use icon only for presets. But in the other hand, using the preset name will not be fine too because it will not fit this design. That's the main reason we didn't add this feature for now. I will discuss that with the team.

Here's an example of my thermostat :
CleanShot 2023-09-19 at 11 07 36

**Comment by Weissnix4711** [comment]
Yup, I realised when messing with it. 

Maybe we could either exclude custom presets, or allow the user to configure an icon for each custom preset?

Only issue with the latter would be how to handle additional presets which turn up after the card has already been configured. We could include them (with no icon, which just doesn't fix anything), or just ignore them and the card must manually be reconfigured for them to show up. Either way, not an ideal solution.

**Comment by piitaya** [comment]
I discussed with the design team and we can not use icon only component for custom presets because some custom presets doesn't have icon.
We will look to add a select dropdown component that fit the tile feature (similar than the one we have in the more info). This component could be used for preset mode, fan mode, swing mode but also for `select` and `input_select`.

We will also look how we can provide icons for custom presets (not only in tile card but for more info too) and then having icon only component for preset mode can make sense.

**Comment by Weissnix4711** [comment]
> we can not use icon only component for custom presets

No worries, makes sense. As a compromise, how about only including presets with known icons (ie. all the default ones) for now, until a better solution is found?

> We will also look how we can provide icons for custom presets (not only in tile card but for more info too)

That would be quite interesting. I feel like the tile cards work so well because of their simplicity, and the other option, a drop down, might undo some of that. I guess it's always better to have something than nothing though, and if the dropdown feature could be reused for other stuff, that's even better.

File: src/panels/lovelace/create-element/create-tile-feature-element.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import "../tile-features/hui-alarm-modes-tile-feature";
 import "../tile-features/hui-climate-hvac-modes-tile-feature";
+import "../tile-features/hui-climate-presets-tile-feature";
+import "../tile-features/hui-target-temperature-tile-feature";
```
Comment (piitaya) [inline src/panels/lovelace/create-element/create-tile-feature-element.ts:4]:
```suggestion
```

File: src/panels/lovelace/editor/config-elements/hui-tile-card-features-editor.ts [hunk 2]
```diff
@@ -47,6 +48,8 @@ type SupportsFeature = (stateObj: HassEntity) => boolean;
 const FEATURE_TYPES: FeatureType[] = [
   "alarm-modes",
   "climate-hvac-modes",
+  "climate-presets",
+  "target-temperature",
```
Comment (piitaya) [inline src/panels/lovelace/editor/config-elements/hui-tile-card-features-editor.ts:52]:
```suggestion
```

**Review by piitaya — APPROVED** [review]

**Comment by piitaya** [comment]
Hello 👋
We added dropdown for `select` and `input_select` feature. I will merge this PR and then do another PR to add support for both style (icons and dropdown). I will also let the user choose which preset to display because preset list can be huge and will not work with icons style.

**Pull request merged**